diff --git a/Makefile b/Makefile
index 1051389..17d395e 100644
--- a/Makefile
+++ b/Makefile
@@ -11,7 +11,7 @@ datarootdir ?= $(prefix)/share
 docdir ?= $(datarootdir)/doc/numakind-$(VERSION)
 initddir ?= /etc/rc.d/init.d
 
-CFLAGS_EXTRA = -fPIC -Wall -Werror -O3
+CFLAGS_EXTRA = -fPIC -Wall -Werror -O0 -g
 OBJECTS = numakind.o numakind_hbw.o hbwmalloc.o
 
 all: libnumakind.so.0.0 numakind-pmtt
diff --git a/test/Makefile b/test/Makefile
index caadd23..7879169 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -1,5 +1,5 @@
 CC=g++
-CFLAGS=-O3 -Wall -fopenmp -std=c++0x
+CFLAGS=-O0 -g -Wall -fopenmp -std=c++0x
 JEMALLOC_PREFIX ?= /usr
 NUMAKIND_PREFIX ?= /usr
 INCLUDE = -I$(JEMALLOC_PREFIX)/include -I$(NUMAKIND_PREFIX)/include
diff --git a/test/bin/test.sh b/test/bin/test.sh
index b753459..ff2d654 100755
--- a/test/bin/test.sh
+++ b/test/bin/test.sh
@@ -32,15 +32,20 @@ then
     rm askpass.sh
  fi
 
-scp ~/rpmbuild/RPMS/x86_64/numakind-0.0*.rpm mic@$knlmcdram:
-scp ~/rpmbuild/RPMS/x86_64/jemalloc-3.5*.rpm mic@$knlmcdram:
+scp ~/rpmbuild/RPMS/x86_64/numakind*.rpm mic@$knlmcdram:
+scp ~/rpmbuild/RPMS/x86_64/jemalloc*.rpm mic@$knlmcdram:
 scp /usr/lib64/libgtest*.so.0.0.0 mic@$knlmcdram:
 scp $basedir/all_tests mic@$knlmcdram:
 ssh root@$knlmcdram "rpm -e numakind >& /dev/null"
 ssh root@$knlmcdram "rpm -e jemalloc >& /dev/null"
-ssh root@$knlmcdram "rpm -i ~mic/jemalloc-3.5*.rpm ~mic/numakind-0.0*.rpm"
+ssh root@$knlmcdram "rpm -e numakind-debuginfo >& /dev/null"
+ssh root@$knlmcdram "rpm -e jemalloc-debuginfo >& /dev/null"
+ssh root@$knlmcdram "rpm -i ~mic/jemalloc*.rpm ~mic/numakind*.rpm"
 ssh root@$knlmcdram "install ~mic/libgtest*.so.0.0.0 /usr/lib64"
 ssh root@$knlmcdram "/sbin/ldconfig"
+
+exit
+
 ssh mic@$knlmcdram "./all_tests --gtest_output=xml:all_tests.xml"
 scp mic@$knlmcdram:all_tests.xml .
 if [ -f simics.pid ]
